garden/ros2_gazebo_integration_tutorial.md

```diff
@@ -0,0 +1,113 @@
+
+Note: The full source code for this tutorial can be found in the [ros_gz_example_brigup package](https://github.com/gazebosim/ros_gz_project_template/tree/main/ros_gz_example_bringup/launch) launch files.
+
+## Implementation
+
+### Load robot description to the parameter server
+
+Load the robot description file to set the `robot_description` parameter on the parameter server.
+
+```python
+    sdf_file  =  os.path.join(pkg_project_description, 'models', 'rrbot', 'model.sdf')
+    with open(sdf_file, 'r') as infp:
+        robot_desc = infp.read()
+```
+
+### Launch state publisher nodes
+
+For publishing and controlling the robot pose, we need joint states of the robot.
+
+![jsp](tutorials/ros2_integration/jsp_diagram.jpg)
+
+1. The `joint_state_publisher` reads the `robot_description` parameter from the parameter server, finds all of the non-fixed joints and publishes a [JointState](https://docs.ros.org/en/api/sensor_msgs/html/msg/JointState.html) message with all those joints defined.
+
+```python
+    joint_state_publisher_gui = Node(
+        package='joint_state_publisher_gui',
+        executable='joint_state_publisher_gui',
+        name='joint_state_publisher_gui',
+        arguments=[sdf_file],
+        output=['screen']
+    )
+```
+1. Visualize in RViz and with the help of the `joint_state_publisher_gui`, configure your robot model.
+See [documentation](https://wiki.ros.org/joint_state_publisher?distro=noetic) for node API.
+
+This functionality is useful during initial development of the model. Afterwards, once you've set up simulation, it might be less useful.
+At this point we have achieved the first aim defined in [Setup](#Setup). 
+
+
+1. Now if you'd want to extend this to visualize robot motion, we need positions and transforms.
+The `robot_state_publisher` takes the description and joint angles of the robot as inputs and publishes the 3D poses of the robot links, using a kinematic tree model of the robot. 
+
+```python
+    robot_state_publisher = Node(
+        package='robot_state_publisher',
+        executable='robot_state_publisher',
+        name='robot_state_publisher',
+        output='both',
+        parameters=[
+            {'use_sim_time': True},
+            {'robot_description': robot_desc},
+        ]
+    )
+```
+
+These 3D poses are published on the `/tf` topic which is useful to track/plan robot's motion, communicate between different parts of the robot or evaluate robot's performance.
+
+### Configure a communication bridge
+
+These joint states can either come from `joint_state_publisher` as seen earlier or from simulated [JointStatePub](https://gazebosim.org/api/gazebo/4.5/classignition_1_1gazebo_1_1systems_1_1JointStatePublisher.html#details) system's joint_state message.
+
+Configure a bridge between ROS topic `/joint_states` and Gazebo topic `/world/demo/model/diff_drive/joint_state` by adding remappings in the node setup or by creating a bridge.yaml: 
```

Comment:
Where is the bridge.yaml file used or read? The linked page (ROS 2 Integration) doesn't say anything about a YAML file. Maybe add a link to a page that does?